lease/leasehttp/http.go

```diff
@@ -50,64 +50,93 @@ func (h *leaseHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	ttl, err := h.l.Renew(lease.LeaseID(lreq.ID))
-	if err != nil {
-		if err == lease.ErrLeaseNotFound {
-			http.Error(w, err.Error(), http.StatusNotFound)
+	switch lreq.Probe {
+	case true:
+		le := h.l.Lookup(lease.LeaseID(lreq.ID))
+		if le == nil {
+			http.Error(w, lease.ErrLeaseNotFound.Error(), http.StatusNotFound)
 			return
 		}
-
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
-	}
-
-	// TODO: fill out ResponseHeader
-	resp := &pb.LeaseKeepAliveResponse{ID: lreq.ID, TTL: ttl}
-	v, err := resp.Marshal()
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+		// TODO: fill out ResponseHeader
+		resp := &pb.LeaseKeepAliveResponse{ID: lreq.ID, TTL: le.TTL, Remaining: int64(le.GetRemaining().Seconds()), Keys: le.GetKeys()}
+		v, err := resp.Marshal()
```

Comment:
this code down to `w.Write(v)` is common to both paths; should move to after the if/else to deduplicate